core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -98,8 +101,11 @@
    * stream reads from a remote worker.
    */
   protected BlockOutStream mCurrentCacheStream;
-  /** The blockId used in the block streams. */
-  private long mStreamBlockId;
```

Comment:
why we remove this and use the `mBlockIdIndex` to indirectly retrieves this value?
can we have the following as a set of states:

```java
private long mPos;
private long mCurrentBlockId;
private long mCurrentBlockStartPos; // starting offset in bytes in file for the current block
private long mCurrentBlockEndPos;  // ending offset in bytes in file for the current block
```
Can we keep all above four metrics in sync, and avoid division to check whether the cache stream is in sync based on `mCurrentBlockStartPos` and `mCurrentBlockEndPos`?

Reply:
after `seek` or `skip`, how can we get the block ID based on these states without division?

Reply:
also, maintaining the state of `mCurrentBlockEndPos` will need extra logic to consider the last block which might not be full.

Reply:
also, even in streaming read, if `mPos` is out of range `[mCurrentBlockStartPos, mCurrentBlockEndPos]`, how to update `mCurrentBlockId`? I think we still need to maintain `mCurrentBlockIdIndex`, or recompute the index based on `mPos / mBlockSize`.

Reply:
@uronce-cc 
Optimization opportunity:
- On streaming read case, when `mPos` is out side of `[mCurrentBlockStartPos, mCurrentBlockEndPos)`, , `mCurrentBlockId++` as that is how we compute the next block within the same file. 
- On seek, division (or shift-opts as I will describe below) is perhaps still needed.

Reply:
I think the cost of comparing `mPos` against range `[mCurrentBlockStartPos, mCurrentBlockEndPos)` might not be a too much optimization over testing `mCurrentStream == null || mCurrentStream.remaining() == 0`. Also, with this approach, we introduce two more states to maintain.